integration-tests/e2e_tests_ci_tool/cmd/check_tests_cmd_test.go

```diff
@@ -0,0 +1,47 @@
+package cmd
+
+import (
+	"testing"
+)
+
+func TestMatchTestNameInCmd(t *testing.T) {
+	tests := []struct {
+		cmd      string
+		testName string
+		expected bool
+	}{
+		{"go test -test.run ^TestExample$", "TestExample", true},
+		{"go test -test.run ^TestExample$", "TestAnother", false},
+		{"go test -test.run ^TestExample$ -v", "TestExample", true},
+		{"go test -test.run ^TestExamplePart$", "TestExample", false},
+		{"go test -test.run ^TestWithNumbers123$", "TestWithNumbers123", true},
+		{"go test -test.run ^Test_With_Underscores$", "Test_With_Underscores", true},
+		{"go test -test.run ^Test-With-Dash$", "Test-With-Dash", true},
+		{"go test -test.run ^TestWithSpace Space$", "TestWithSpace Space", true},
+		{"go test -test.run ^TestWithNewline\nNewline$", "TestWithNewline\nNewline", true},
+		{"go test -test.run ^TestOne$|^TestTwo$", "TestOne", true},
+		{"go test -test.run ^TestOne$|^TestTwo$", "TestTwo", true},
+		{"go test -test.run ^TestOne$|^TestTwo$", "TestThree", false},
+		{"go test -test.run TestOne|TestTwo", "TestTwo", true},
+		{"go test -test.run TestOne|TestTwo", "TestOne", true},
+		{"go test -test.run TestOne|TestTwo|TestThree", "TestFour", false},
+		{"go test -test.run ^TestOne$|TestTwo$", "TestTwo", true},
+		{"go test -test.run ^TestOne$|TestTwo|TestThree$", "TestThree", true},
+		{"go test -test.run TestOne|TestTwo|TestThree", "TestOne", true},
+		{"go test -test.run TestOne|TestTwo|TestThree", "TestThree", true},
+		{"go test -test.run ^TestA$|^TestB$|^TestC$", "TestA", true},
+		{"go test -test.run ^TestA$|^TestB$|^TestC$", "TestB", true},
+		{"go test -test.run ^TestA$|^TestB$|^TestC$", "TestD", false},
+		{"go test -test.run TestA|^TestB$|TestC", "TestB", true},
+		{"go test -test.run ^TestA|^TestB|TestC$", "TestA", true},
+		{"go test -test.run ^TestA|^TestB|TestC$", "TestC", true},
+		{"go test -test.run ^TestA|^TestB|TestC$", "TestD", false},
```

Comment:
same as before, what if I do this: `go test -test.run my_tests.go` or `go test -test.run ^Test[A-C]$"` or just `go test -test.run '.*'`?

Reply:
@Tofel will go test -test.run my_tests.go work? Note that `-test.run` is used to specify test name inside a file.